src/workerd/io/actor-sqlite.c++

```diff
@@ -157,11 +158,68 @@ void ActorSqlite::onWrite() {
       // rather than after the callback.
       { auto drop = kj::mv(txn); }
 
-      return commitCallback();
+      return commitImpl();
     })));
   }
 }
 
+kj::Promise<void> ActorSqlite::commitImpl() {
+  bool needsAlarmFlush = false;
+  kj::Maybe<kj::Date> newAlarmTime;
+  KJ_SWITCH_ONEOF(currentAlarmTime) {
+    KJ_CASE_ONEOF(_, UnknownAlarmTime) {
+      // Haven't tried to write alarm yet, so don't need to flush.
+    }
+    KJ_CASE_ONEOF(knownAlarmTime, KnownAlarmTime) {
+      if (knownAlarmTime.status == KnownAlarmTime::Status::DIRTY) {
+        knownAlarmTime.status = KnownAlarmTime::Status::FLUSHING;
+        needsAlarmFlush = true;
+        newAlarmTime = knownAlarmTime.time;
+      }
+    }
+    KJ_CASE_ONEOF(deferredDelete, DeferredAlarmDelete) {
+      if (deferredDelete.status == DeferredAlarmDelete::Status::READY) {
+        deferredDelete.status = DeferredAlarmDelete::Status::FLUSHING;
+        needsAlarmFlush = true;
+        newAlarmTime = kj::none;
+      }
+    }
+  }
+
+  // TODO(soon): ActorCache implements a 4x retry of failed flushes... Do we need anything similar
```

Comment:
Probably, yes. It's possible that the update will throw DISCONNECTED if it was sent to the server over a connection that was lost during the call. We need to retry in that case.

However, arguably, it could be the hook's responsibility to perform the retry, rather than do it here. In the workerd case, where the hook is implemented entirely locally, there'd be no reason to retry.

Reply:
Seconded. We need retries somewhere.